activation/activation.go

```diff
@@ -483,58 +508,63 @@ func (b *Builder) Coinbase() types.Address {
 	return b.coinbaseAccount
 }
 
-// MinGas [...].
-func (b *Builder) MinGas() uint64 {
-	panic("not implemented")
-}
-
-// SetMinGas [...].
-func (b *Builder) SetMinGas(value uint64) {
-	panic("not implemented")
-}
-
-func getNIPostKey() []byte {
-	return []byte("NIPost")
-}
-
-func (b *Builder) storeChallenge(ch *types.NIPostChallenge) error {
-	bts, err := codec.Encode(ch)
+func (b *Builder) loadChallenge() error {
+	nipost, err := store.GetNIPostChallenge(b.cdb)
 	if err != nil {
-		return fmt.Errorf("serialize NIPost challenge: %w", err)
+		return err
 	}
+	b.challenge = nipost
+	return nil
+}
 
-	if err := niposts.Add(b.cdb, getNIPostKey(), bts); err != nil {
-		return fmt.Errorf("put NIPost challenge to database: %w", err)
+func (b *Builder) discardChallenge() {
+	b.challenge = nil
+	b.pendingATX = nil
+	if err := store.ClearNIPostChallenge(b.cdb); err != nil {
+		b.log.Error("failed to discard NIPost challenge: %w", err)
 	}
-
-	return nil
 }
 
-func (b *Builder) loadChallenge() error {
-	bts, err := niposts.Get(b.cdb, getNIPostKey())
-	if err != nil {
-		return fmt.Errorf("get NIPost challenge from store: %w", err)
+func (b *Builder) getCommitmentAtx(ctx context.Context) (*types.ATXID, error) {
```

Comment:
how do you ensure that
- all atxs of the same node id uses the same commitment atx
- if identity file change (node id change), this commitment cache is cleared

Reply:
Add 1): For every ATX a node receives it fetches the very first ATX published by it and verifies that the NiPost still uses the same commitment ATX.

Add 2): Does the identity of the node change while it is running? Isn't a re-creation of PoST necessary when the identity changes? This would also trigger a new initial ATX which would then set a new commitment ATX for the node.

This method `getCommitmentAtx` only gets the commitment ATX for the node itself, not for other nodes.